src/windows/main_window.py

```diff
@@ -153,6 +156,9 @@ def closeEvent(self, event):
         # Destroy lock file
         self.destroy_lock_file()
 
+        # Main window widget closed
+        log.info('================================================\n')
```

Comment:
Personally, I'm good with this _except_ for the newline at the end of this line. I just don't think a blank line in the logfile is very helpful, especially since (as you say) other threads from the session can still log messages **after** that line.

If we're going to insert newline separators to signify session boundaries in the logfile, I'd rather insert one **BEFORE** the _first_ log message from a NEW session. That way, we can unequivocally guarantee that the blank line is the exact division between two sessions.

As far as the end-of-shutdown marker, either logging a line of `====` twice in a row, or logging something like `======= Main thread terminating =======` would be my preference, over the line with an extra newline appended. 

That way, if any further lines do follow that marker, it'll be clear that they come from background threads but still belong to the preceding session. (Especially if we do the newline-before-the-first-message thing.) And its absence will be a clear indicator of abnormal shutdowns, should something crash inside the `closeEvent()` handler.

Reply:
(In fact, while you're in there...)

Since the logfiles don't have timestamps, I've often thought that logging just the start and end time of each session would be a useful thing. So, as part of that session-boundary logging, maybe the first and last-but-for-stragglers log messages could be something like this:

```py
# Start
import time
now = time.asctime()
log.info('\n================================================')
log.info('=========' + now.center(30) + '=========')

#End
import time
now = time.asctime()
log.info('=========' + now.center(30) + '=========')
log.info('================================================')
```